diff --git a/libbcachefs.c b/libbcachefs.c
index df66514..e84a111 100644
--- a/libbcachefs.c
+++ b/libbcachefs.c
@@ -953,7 +953,7 @@ int bchu_data(struct bchfs_handle fs, struct bch_ioctl_data cmd)
                               e.p.pos.inode,
                               e.p.pos.offset);
                }
-
+               fflush(stdout);
                sleep(1);
        }
        printf("\nDone\n");
